feedback-mailer/feedback-mailer-from-sheet.gs

```diff
@@ -0,0 +1,473 @@
+  } catch (error) {
+    return false;
+  }
+}
+
+function sendEmail(studentEmail, mailBody) {
+  let status = "";
+  let errorMsg = "";
+  if (!validateEmail(studentEmail)) {
+    status = MAIL_STATUS_FAILED;
+    errorMsg = `Invalid email: ${studentEmail}`;
+    return { status , errorMsg };
+  }
+  try {
+    GmailApp.sendEmail(studentEmail, MAIL_SUBJECT, mailBody, { name: OWNER_NAME , htmlBody : mailBody , bcc : BCC_EMAIL } );
+    status = MAIL_STATUS_SENT;
+  } catch (error) {
+    status = MAIL_STATUS_FAILED;
+    errorMsg = "Mail sending failed. " + error.toString();
+  }
+  return { status , errorMsg };
+}
+
+function sendMailsToStudents() {
+
+  let setupResult = initSetup();
+  if (!setupResult.success) {
+    return;
+  }
+
+  if (!confirmSendingMail(Session.getActiveUser().getEmail())) {
+    showInfo(TITLE_ERROR, MSG_MAIL_REJECTED);
+    return;
+  }
+
+  let { activeSheet, feedbackData } = setupResult;
+
+  let errorRows = [];
+  let skippedRows = [];
+  let successRows = [];
+
+  for (let rowIndex = 0; rowIndex < feedbackData.length; ++rowIndex) {
+    let rowNumber = rowIndex + DATA_BEGIN_ROW;
+
+    let mailSentStatusCell = activeSheet.getRange(rowNumber, INDEX_MAIL_STATUS + 1);
+    if (mailSentStatusCell.getValue().trim() === MAIL_STATUS_SENT) {
+      // Mail is already sent
+      skippedRows.push(rowNumber);
+      continue;
+    }
+
+    let studentFormFeedback = {}
+    studentFormFeedback[FORM_TIMESTAMP] = feedbackData[rowIndex][INDEX_TIMESTAMP];
+    studentFormFeedback[FORM_EMAIL_ADDRESS] = feedbackData[rowIndex][INDEX_EMAIL_ADDRESS];
+    studentFormFeedback[FORM_STUDENT_NAME] = feedbackData[rowIndex][INDEX_STUDENT_NAME];
+    studentFormFeedback[FORM_STUDENT_EMAIL] = feedbackData[rowIndex][INDEX_STUDENT_EMAIL];
+    studentFormFeedback[FORM_SSM_EMAIL] = feedbackData[rowIndex][INDEX_SSM_EMAIL];
+    studentFormFeedback[FORM_ALLOTTED_PROJECT] = feedbackData[rowIndex][INDEX_ALLOTTED_PROJECT];
+    studentFormFeedback[FORM_DISCUSSED_PROJECT] = feedbackData[rowIndex][INDEX_DISCUSSED_PROJECT];
+    studentFormFeedback[FORM_VIDEO_STATUS] = feedbackData[rowIndex][INDEX_VIDEO_STATUS];
+    studentFormFeedback[FORM_VIDEO_DURATION] = feedbackData[rowIndex][INDEX_VIDEO_DURATION];
+    studentFormFeedback[FORM_SPEECH_APPEARANCE] = feedbackData[rowIndex][INDEX_PROJECT_EXPLANATION_SCRIPT];
+    studentFormFeedback[FORM_SPEECH_FLOW] = feedbackData[rowIndex][INDEX_SPEECH_FLOW];
+    studentFormFeedback[FORM_GENERAL_FEATURES] = feedbackData[rowIndex][INDEX_FEATURES];
+    studentFormFeedback[FORM_TECH_STACK] = feedbackData[rowIndex][INDEX_TECH_STACK];
+    studentFormFeedback[FORM_SELF_CONTRIBUTION] = feedbackData[rowIndex][INDEX_SELF_CONTRIBUTION];
+
+    let { mailBody , score } = createMailBody(studentFormFeedback);
+
+    let scoreCell = activeSheet.getRange(rowNumber, INDEX_SCORE + 1);
+    scoreCell.setValue(score);
+    let { status , errorMsg } = sendEmail(studentFormFeedback[FORM_STUDENT_EMAIL].trim(), mailBody);
```

Comment:
Please trim while constructing `studentFormFeedback`. Trim all the values.

Reply:
Updated the changes in next commit